### PR TITLE
Update Content Security Policy for Social Embeds

### LINKED ISSUE #6037: CSP changes for social embeds
**Is your feature request related to a problem? Please describe.**
To be able to display social embeds on our pages involves loading third-party resources including scripts and images from each social network.

We need to adapt our Content Security Policy (CSP) to support the required functionality while  maintaining a high degree of protection.

**Parts**
1. https://github.com/bbc/simorgh/pull/6209 ✅
2. https://github.com/bbc/mozart-routing/pull/1307  ✅

**Describe the solution you'd like**
- identify the most restrictive set of CSP directives that supports our social embeds requirements
- implement and validate these changes in the Mozart codebase

**Describe alternatives you've considered**
N/A

**Testing notes**
[Tester to complete]

Dev insight: Will Cypress tests be required or are unit tests sufficient? Will there be any potential regression? etc

- [ ] This feature is expected to need manual testing.

**Additional context**
Related/relevant issues:
- https://github.com/bbc/simorgh/issues/5922
- https://github.com/bbc/simorgh/issues/5391

### PR DESCRIPTION
**Pull request by simonsinclair**
Resolves https://github.com/bbc/simorgh/issues/6037.

## Overall change:
Add CSP directives for Social Embeds.

## Code changes:
- Update `utilities/cspHeader/index.js`
- Update `utilities/cspHeader/index.test.js`

## Method
1. `npm build && npm start`
2. Visit:
    - Canonical: http://localhost:7080/mundo/embeds-23263889
    - AMP: http://localhost:7080/mundo/embeds-23263889.amp
3. Log CSP errors relating to Social Embeds.
4. Add directives.

Rinse and repeat until errors are nullified.

### Final CSP errors for Canonical
_Implementation uses oEmbed._
```
Refused to load the script 'https://platform.twitter.com/widgets.js' because it violates the following Content Security Policy directive: "script-src https://news.files.bbci.co.uk https://news.test.files.bbci.co.uk https://*.chartbeat.com https://*.go-mpulse.net https://mybbc-analytics.files.bbci.co.uk https://emp.bbci.co.uk https://static.bbci.co.uk http://*.chartbeat.com http://localhost:1124 'self' 'unsafe-inline'". Note that 'script-src-elem' was not explicitly set, so 'script-src' is used as a fallback.
Refused to load the script 'https://www.instagram.com/embed.js' because it violates the following Content Security Policy directive: "script-src https://news.files.bbci.co.uk https://news.test.files.bbci.co.uk https://*.chartbeat.com https://*.go-mpulse.net https://mybbc-analytics.files.bbci.co.uk https://emp.bbci.co.uk https://static.bbci.co.uk http://*.chartbeat.com http://localhost:1124 'self' 'unsafe-inline'". Note that 'script-src-elem' was not explicitly set, so 'script-src' is used as a fallback.
Refused to load the script 'http://www.instagram.com/embed.js' because it violates the following Content Security Policy directive: "script-src https://news.files.bbci.co.uk https://news.test.files.bbci.co.uk https://*.chartbeat.com https://*.go-mpulse.net https://mybbc-analytics.files.bbci.co.uk https://emp.bbci.co.uk https://static.bbci.co.uk http://*.chartbeat.com http://localhost:1124 'self' 'unsafe-inline'". Note that 'script-src-elem' was not explicitly set, so 'script-src' is used as a fallback.
Refused to load the script 'https://cdn.syndication.twimg.com/tweets.json?features=tfw_horizon_tweet_embed_9555%3Acontrol&callback=__twttr.callbacks.cb0&ids=1237210910835392512&lang=es&suppress_response_codes=true&theme=light&tz=GMT%2B0100' because it violates the following Content Security Policy directive: "script-src https://news.files.bbci.co.uk https://news.test.files.bbci.co.uk https://*.chartbeat.com https://*.go-mpulse.net https://mybbc-analytics.files.bbci.co.uk https://emp.bbci.co.uk https://static.bbci.co.uk http://*.chartbeat.com http://localhost:1124 https://platform.twitter.com https://www.instagram.com 'self' 'unsafe-inline'". Note that 'script-src-elem' was not explicitly set, so 'script-src' is used as a fallback.

Refused to frame 'https://www.youtube.com/' because it violates the following Content Security Policy directive: "frame-src 'self' https://polling.bbc.co.uk https://polling.test.bbc.co.uk https://securepubads.g.doubleclick.net https://tpc.googlesyndication.com https://emp.bbc.com https://emp.bbc.co.uk https://chartbeat.com https://*.chartbeat.com".
Refused to frame 'https://platform.twitter.com/' because it violates the following Content Security Policy directive: "frame-src 'self' https://polling.bbc.co.uk https://polling.test.bbc.co.uk https://securepubads.g.doubleclick.net https://tpc.googlesyndication.com https://emp.bbc.com https://emp.bbc.co.uk https://chartbeat.com https://*.chartbeat.com https://www.youtube.com".
Refused to frame 'https://syndication.twitter.com/' because it violates the following Content Security Policy directive: "frame-src 'self' https://polling.bbc.co.uk https://polling.test.bbc.co.uk https://securepubads.g.doubleclick.net https://tpc.googlesyndication.com https://emp.bbc.com https://emp.bbc.co.uk https://chartbeat.com https://*.chartbeat.com https://www.youtube.com https://platform.twitter.com https://www.instagram.com".

Refused to load the stylesheet 'https://platform.twitter.com/css/tweet.d41c1d7e4bac44f4658ca45d09564e79.light.ltr.css' because it violates the following Content Security Policy directive: "style-src https://news.files.bbci.co.uk 'unsafe-inline'". Note that 'style-src-elem' was not explicitly set, so 'style-src' is used as a fallback.

Refused to load the image 'https://platform.twitter.com/css/tweet.d41c1d7e4bac44f4658ca45d09564e79.light.ltr.css' because it violates the following Content Security Policy directive: "img-src https://ichef.bbci.co.uk https://ichef.test.bbci.co.uk https://ping.chartbeat.net https://a1.api.bbc.co.uk/hit.xiti https://logws1363.ati-host.net https://news.files.bbci.co.uk https://news.test.files.bbci.co.uk https://*.akstat.io https://r.bbci.co.uk https://pagead2.googlesyndication.com https://securepubads.g.doubleclick.net https://tpc.googlesyndication.com https://www.google.com https://via.placeholder.com http://b.files.bbci.co.uk http://ping.chartbeat.net https://syndication.twitter.com data: 'self'".
Refused to load the image 'https://syndication.twitter.com/i/jot?l=%7B%22widget_origin%22%3A%22http%3A%2F%2Flocalhost%3A7080%2Fmundo%2Fembeds-23263889%22%2C%22widget_frame%22%3Afalse%2C%22widget_site_screen_name%22%3A%22bbcmundo%22%2C%22widget_creator_screen_name%22%3A%22bbcmundo%22%2C%22item_ids%22%3A%5B%221237210910835392512%22%5D%2C%22item_details%22%3A%7B%221237210910835392512%22%3A%7B%22item_type%22%3A0%7D%7D%2C%22_category_%22%3A%22tfw_client_event%22%2C%22triggered_on%22%3A1587040052017%2C%22dnt%22%3Afalse%2C%22client_version%22%3A%22a2aa63b%3A1586240908307%22%2C%22format_version%22%3A1%2C%22event_namespace%22%3A%7B%22client%22%3A%22tfw%22%2C%22page%22%3A%22tweet%22%2C%22section%22%3A%22subject%22%2C%22component%22%3A%22rawembedcode%22%2C%22action%22%3A%22no_result%22%7D%7D' because it violates the following Content Security Policy directive: "img-src https://ichef.bbci.co.uk https://ichef.test.bbci.co.uk https://ping.chartbeat.net https://a1.api.bbc.co.uk/hit.xiti https://logws1363.ati-host.net https://news.files.bbci.co.uk https://news.test.files.bbci.co.uk https://*.akstat.io https://r.bbci.co.uk https://pagead2.googlesyndication.com https://securepubads.g.doubleclick.net https://tpc.googlesyndication.com https://www.google.com https://via.placeholder.com http://b.files.bbci.co.uk http://ping.chartbeat.net data: 'self'".
Refused to load the image 'https://pbs.twimg.com/profile_images/1241865031001812992/oqzQEDGR_bigger.jpg' because it violates the following Content Security Policy directive: "img-src https://ichef.bbci.co.uk https://ichef.test.bbci.co.uk https://ping.chartbeat.net https://a1.api.bbc.co.uk/hit.xiti https://logws1363.ati-host.net https://news.files.bbci.co.uk https://news.test.files.bbci.co.uk https://*.akstat.io https://r.bbci.co.uk https://pagead2.googlesyndication.com https://securepubads.g.doubleclick.net https://tpc.googlesyndication.com https://www.google.com https://via.placeholder.com http://b.files.bbci.co.uk http://ping.chartbeat.net https://syndication.twitter.com data: 'self'".
Refused to load the image 'https://i.ytimg.com/vi_webp/chiWVxreqhU/mqdefault.webp' because it violates the following Content Security Policy directive: "img-src https://ichef.bbci.co.uk https://ichef.test.bbci.co.uk https://ping.chartbeat.net https://a1.api.bbc.co.uk/hit.xiti https://logws1363.ati-host.net https://news.files.bbci.co.uk https://news.test.files.bbci.co.uk https://*.akstat.io https://r.bbci.co.uk https://pagead2.googlesyndication.com https://securepubads.g.doubleclick.net https://tpc.googlesyndication.com https://www.google.com https://via.placeholder.com http://b.files.bbci.co.uk http://ping.chartbeat.net https://syndication.twitter.com https://platform.twitter.com https://pbs.twimg.com data: 'self'".
```

### Final CSP errors for AMP
_Implementation uses `amp-twitter`, `amp-youtube`, and `amp-instagram`._
```
Refused to load the image 'https://www.instagram.com/p/B8FPf4ZphHi/media/?size=l' because it violates the following Content Security Policy directive: "img-src https://ichef.bbci.co.uk https://ichef.test.bbci.co.uk https://ping.chartbeat.net https://a1.api.bbc.co.uk/hit.xiti https://logws1363.ati-host.net https://news.files.bbci.co.uk https://news.test.files.bbci.co.uk https://*.akstat.io https://r.bbci.co.uk https://pagead2.googlesyndication.com https://securepubads.g.doubleclick.net https://tpc.googlesyndication.com https://www.google.com https://via.placeholder.com http://b.files.bbci.co.uk http://ping.chartbeat.net https://syndication.twitter.com https://platform.twitter.com https://pbs.twimg.com data: 'self'".
Refused to load the image 'https://i.ytimg.com/vi/chiWVxreqhU/sddefault.jpg' because it violates the following Content Security Policy directive: "img-src https://ichef.bbci.co.uk https://ichef.test.bbci.co.uk https://ping.chartbeat.net https://a1.api.bbc.co.uk/hit.xiti https://logws1363.ati-host.net https://news.files.bbci.co.uk https://news.test.files.bbci.co.uk https://*.akstat.io https://r.bbci.co.uk https://pagead2.googlesyndication.com https://securepubads.g.doubleclick.net https://tpc.googlesyndication.com https://www.google.com https://via.placeholder.com http://b.files.bbci.co.uk http://ping.chartbeat.net https://syndication.twitter.com https://platform.twitter.com https://pbs.twimg.com data: 'self'".
Refused to load the image 'https://i.ytimg.com/vi/chiWVxreqhU/hqdefault.jpg' because it violates the following Content Security Policy directive: "img-src https://ichef.bbci.co.uk https://ichef.test.bbci.co.uk https://ping.chartbeat.net https://a1.api.bbc.co.uk/hit.xiti https://logws1363.ati-host.net https://news.files.bbci.co.uk https://news.test.files.bbci.co.uk https://*.akstat.io https://r.bbci.co.uk https://pagead2.googlesyndication.com https://securepubads.g.doubleclick.net https://tpc.googlesyndication.com https://www.google.com https://via.placeholder.com http://b.files.bbci.co.uk http://ping.chartbeat.net https://syndication.twitter.com https://platform.twitter.com https://pbs.twimg.com data: 'self'".
Refused to load the image 'https://scontent-lhr8-1.cdninstagram.com/v/t51.2885-15/e35/83773347_141521066875658_5767189503154469737_n.jpg?_nc_ht=scontent-lhr8-1.cdninstagram.com&_nc_cat=102&_nc_ohc=VG6IGvu4WwwAX9DJm9C&oh=7ab60c3672fd6b595111bc6f8ba330d6&oe=5EC16465' because it violates the following Content Security Policy directive: "img-src https://ichef.bbci.co.uk https://ichef.test.bbci.co.uk https://ping.chartbeat.net https://a1.api.bbc.co.uk/hit.xiti https://logws1363.ati-host.net https://news.files.bbci.co.uk https://news.test.files.bbci.co.uk https://*.akstat.io https://r.bbci.co.uk https://pagead2.googlesyndication.com https://securepubads.g.doubleclick.net https://tpc.googlesyndication.com https://www.google.com https://via.placeholder.com http://b.files.bbci.co.uk http://ping.chartbeat.net https://i.ytimg.com https://www.instagram.com data: 'self'".

Refused to connect to 'https://d-1461173749120232071.ampproject.net/2004142326360/frame.html' because it violates the following Content Security Policy directive: "connect-src https://*.akstat.io https://*.akamaihd.net https://c.go-mpulse.net https://adservice.google.com https://securepubads.g.doubleclick.net https://pagead2.googlesyndication.com https://tpc.googlesyndication.com https://cdn.ampproject.org https://amp-error-reporting.appspot.com https://logws1363.ati-host.net https://toggles.test.api.bbci.co.uk https://www.bbc.co.uk 'self'".
Refused to connect to 'https://d-10299642461890122076.ampproject.net/2004030010070/frame.html' because it violates the following Content Security Policy directive: "connect-src https://*.akstat.io https://*.akamaihd.net https://c.go-mpulse.net https://adservice.google.com https://securepubads.g.doubleclick.net https://pagead2.googlesyndication.com https://tpc.googlesyndication.com https://cdn.ampproject.org https://amp-error-reporting.appspot.com https://logws1363.ati-host.net https://toggles.test.api.bbci.co.uk https://www.bbc.co.uk 'self'".
Refused to connect to 'https://3p.ampproject.net/2004142326360/f.js' because it violates the following Content Security Policy directive: "connect-src https://*.akstat.io https://*.akamaihd.net https://c.go-mpulse.net https://adservice.google.com https://securepubads.g.doubleclick.net https://pagead2.googlesyndication.com https://tpc.googlesyndication.com https://cdn.ampproject.org https://amp-error-reporting.appspot.com https://logws1363.ati-host.net https://toggles.test.api.bbci.co.uk https://www.bbc.co.uk 'self'".
Refused to connect to 'https://3p.ampproject.net/2004030010070/f.js' because it violates the following Content Security Policy directive: "connect-src https://*.akstat.io https://*.akamaihd.net https://c.go-mpulse.net https://adservice.google.com https://securepubads.g.doubleclick.net https://pagead2.googlesyndication.com https://tpc.googlesyndication.com https://cdn.ampproject.org https://amp-error-reporting.appspot.com https://logws1363.ati-host.net https://toggles.test.api.bbci.co.uk https://www.bbc.co.uk 'self'".
Refused to connect to 'https://platform.twitter.com/widgets.js' because it violates the following Content Security Policy directive: "connect-src https://*.akstat.io https://*.akamaihd.net https://c.go-mpulse.net https://adservice.google.com https://securepubads.g.doubleclick.net https://pagead2.googlesyndication.com https://tpc.googlesyndication.com https://cdn.ampproject.org https://amp-error-reporting.appspot.com https://logws1363.ati-host.net https://toggles.test.api.bbci.co.uk https://www.bbc.co.uk 'self'".

Refused to frame 'https://d-1461173749120232071.ampproject.net/' because it violates the following Content Security Policy directive: "frame-src 'self' https://polling.bbc.co.uk https://polling.test.bbc.co.uk https://securepubads.g.doubleclick.net https://tpc.googlesyndication.com".
Refused to frame 'https://www.youtube.com/' because it violates the following Content Security Policy directive: "frame-src 'self' https://polling.bbc.co.uk https://polling.test.bbc.co.uk https://securepubads.g.doubleclick.net https://tpc.googlesyndication.com".
Refused to frame 'https://www.instagram.com/' because it violates the following Content Security Policy directive: "frame-src 'self' https://polling.bbc.co.uk https://polling.test.bbc.co.uk https://securepubads.g.doubleclick.net https://tpc.googlesyndication.com".
Refused to frame 'https://d-1269826772363750030.ampproject.net/' because it violates the following Content Security Policy directive: "frame-src 'self' https://polling.bbc.co.uk https://polling.test.bbc.co.uk https://securepubads.g.doubleclick.net https://tpc.googlesyndication.com".
Refused to frame 'https://www.youtube.com/' because it violates the following Content Security Policy directive: "frame-src 'self' https://polling.bbc.co.uk https://polling.test.bbc.co.uk https://securepubads.g.doubleclick.net https://tpc.googlesyndication.com".
Refused to frame 'https://d-1269826772363750030.ampproject.net/' because it violates the following Content Security Policy directive: "frame-src 'self' https://polling.bbc.co.uk https://polling.test.bbc.co.uk https://securepubads.g.doubleclick.net https://tpc.googlesyndication.com".
Refused to frame 'https://www.instagram.com/' because it violates the following Content Security Policy directive: "frame-src 'self' https://polling.bbc.co.uk https://polling.test.bbc.co.uk https://securepubads.g.doubleclick.net https://tpc.googlesyndication.com".
```

---

- [x] I have assigned myself to this PR and the corresponding issues
- [x] I have added labels to this PR for the relevant pod(s) affected by these changes
- [x] I have assigned this PR to the Simorgh project

## Testing:

- [x] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [x] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false npm run test:e2e:interactive`)
- [x] This PR requires manual testing

### CODE REVIEW

**Comment by oluoluoxenfree** [comment]
```
Update utilities/cspHeader/index.test.js WIP
```

Is this still WIP?

**Comment by simonsinclair** [comment]
> ```
> Update utilities/cspHeader/index.test.js WIP
> ```
> 
> Is this still WIP?

All done now, @oluoluoxenfree. 😄

File: src/server/utilities/cspHeader/index.js [hunk 1]
```diff
@@ -76,6 +80,9 @@ const directives = {
       'https://polling.bbc.co.uk', // Media page
       'https://securepubads.g.doubleclick.net', // ads
       'https://tpc.googlesyndication.com', // ads
+      'https://www.youtube.com', // Social Embeds, <amp-youtube />
+      'https://www.instagram.com', // Social Embeds, <amp-instagram />
+      'https://*.ampproject.net', // Social Embeds
```
Comment (sareh) [inline src/server/utilities/cspHeader/index.js:85]: This array is for the Canonical page on Test, so you don't need this amp domain here
```suggestion
```

File: src/server/utilities/cspHeader/index.js [hunk 2]
```diff
@@ -86,13 +93,20 @@ const directives = {
...
       "'self'",
     ],
     ampNonLive: [
       'https://polling.bbc.co.uk', // Media page
       'https://polling.test.bbc.co.uk', // Media page
       'https://securepubads.g.doubleclick.net', // ads
       'https://tpc.googlesyndication.com', // ads
+      'https://www.youtube.com', // Social Embeds, <amp-youtube />
+      'https://www.instagram.com', // Social Embeds, <amp-instagram />
+      'https://*.ampproject.net', // Social Embeds
```
Comment (sareh) [inline src/server/utilities/cspHeader/index.js:109]: This array is for the Canonical page on Live, so you don't need this amp domain here
```suggestion
```

**Comment by oluoluoxenfree** [comment]
@simonsinclair Does that list correspond to remaining errors or ones that you've fixed?

Running it now so may answer my own question :)

**Comment by simonsinclair** [comment]
> @simonsinclair Does that list correspond to remaining errors or ones that you've fixed?
> 
> Running it now so may answer my own question :)

Both lists correspond to CSP errors that should now not occur when visiting a page containing Social Embeds on this branch, @oluoluoxenfree.

**Review by FK78 — APPROVED** [review]

File: src/server/utilities/cspHeader/index.js [hunk 3]
```diff
@@ -47,10 +49,12 @@ const directives = {
       'https://pagead2.googlesyndication.com', // ads
       'https://tpc.googlesyndication.com', // ads
       'https://cdn.ampproject.org',
```
Comment (HarveyPeachey) [inline src/server/utilities/cspHeader/index.js:51]: Just out of curiosity do we know what the difference is between ampproject.org and ampproject.net?

File: src/server/utilities/cspHeader/index.js [hunk 4]
```diff
@@ -104,12 +118,30 @@ const directives = {
...
       'https://*.chartbeat.com',
+      'https://www.youtube.com', // Social Embeds
+      'https://platform.twitter.com', // Social Embeds
+      'https://www.instagram.com', // Social Embeds
+      'https://syndication.twitter.com', // Social Embeds
       "'self'",
     ],
   },
   imgSrc: {
     ampLive: [
```
Comment (HarveyPeachey) [inline src/server/utilities/cspHeader/index.js:129]: These have now been added because they're now different to canonical?

**Review by HarveyPeachey — COMMENTED** [review]
Looks good just a few questions. Also don't think I have enough knowledge about CSP to approve.

File: src/server/utilities/cspHeader/index.js (hunk 4 above)
Comment (simonsinclair) [reply src/server/utilities/cspHeader/index.js:129]: Yes, that's right. AMP Social Embeds use a different implementation to Canonical.

File: src/server/utilities/cspHeader/index.js (hunk 3 above)
Comment (simonsinclair) [reply src/server/utilities/cspHeader/index.js:51]: I couldn't determine any differences - they're both owned by Google and redirect to amp.dev.

**Review by chris-hinds — APPROVED** [review]

File: src/server/utilities/cspHeader/index.js [hunk 5]
```diff
@@ -19,8 +19,10 @@ const directives = {
       'https://a1.api.bbc.co.uk/hit.xiti', // ATI
       'https://toggles.api.bbci.co.uk', // Toggles service
       'https://cdn.ampproject.org', // AMP
+      'https://*.ampproject.net', // Social Embeds
```
Comment (sareh) [inline src/server/utilities/cspHeader/index.js:22]: Needed since this file is requested from this domain: https://3p.ampproject.net/2004030010070/f.js

**Review by sareh — CHANGES_REQUESTED** [review]
When running with `npm run build` `npm run start` and looking in the console on Chrome, I get these following CSP errors:

AMP
http://localhost:7080/mundo/embeds-23263889.amp
Screenshot CSP errors - AMP

Canonical:
http://localhost:7080/mundo/embeds-23263889
Screenshot CSP errors canonical

`style-src` for non-Live needs `https://news.test.files.bbci.co.uk`
`connect-src` for non-Live needs `https://news.test.files.bbci.co.uk`
`font-src` for non-Live and Live both need `https://static.bbci.co.uk` for legacy fonts for embeds.

**Comment by simonsinclair** [comment]
> When running with `npm run build` `npm run start` and looking in the console on Chrome, I get these following CSP errors:
> 
> AMP
> http://localhost:7080/mundo/embeds-23263889.amp
> Screenshot CSP errors - AMP
> 
> Canonical:
> http://localhost:7080/mundo/embeds-23263889
> Screenshot CSP errors canonical
> 
> `style-src` for non-Live needs `https://news.test.files.bbci.co.uk`
> `connect-src` for non-Live needs `https://news.test.files.bbci.co.uk`
> `font-src` for non-Live and Live both need `https://static.bbci.co.uk` for legacy fonts for embeds.

Thanks for taking a look, @Sareh. This PR fixes errors relating to Social Embeds only, so these errors will not be resolved. They seem to relate to Includes, which is currently a WIP.

File: src/server/utilities/cspHeader/index.js (hunk 5 above)
Comment (simonsinclair) [reply src/server/utilities/cspHeader/index.js:22]: Yes - this file seems to change 'version' quite a bit.

**Comment by PriyaKR** [comment]
I don't see any CSP errors in my console.
Screenshot 2020-04-20 at 10 29 30

Screenshot 2020-04-20 at 10 24 20

**Comment by simonsinclair** [comment]
> I don't see any CSP errors in my console.
> Screenshot 2020-04-20 at 10 23 20
> Screenshot 2020-04-20 at 10 24 20

It looks like that's an `.amp` URL, where Includes are not rendered currently.

**Comment by sareh** [comment]
@PriyaKR It seems those errors only occur when you scroll down to the bottom of the page - since the embeds only then try to fetch fonts/styles for those pages. 

Since @simonsinclair has explained that the Visual Journalism includes is still in-progress, I'm happy to have this merged in and have the VJ CSP headers added alongside that work.

**Review by sareh — APPROVED** [review]

**Comment by PriyaKR** [comment]
> @PriyaKR It seems those errors only occur when you scroll down to the bottom of the page - since the embeds only then try to fetch fonts/styles for those pages.
> 
> Since @simonsinclair has explained that the Visual Journalism includes is still in-progress, I'm happy to have this merged in and have the VJ CSP headers added alongside that work.

I don't see them if i scroll past the page.

**Comment by PriyaKR** [comment]
No CSP errors related to social embeds seen on console.But found the social embeds are not being displayed on amp pages on IE11 issue seen on latest too @paruchurisilpa to raise an issue.

**Pull request merged**